brainstat/tests/datagen_peak_clus.py

```diff
@@ -0,0 +1,217 @@
+    print("test data %s for peak_clus generated... " % (str(test_num)))
+
+    with open(fin_name, "wb") as f:
+        pickle.dump(I, f, protocol=4)
+    with open(fout_name, "wb") as g:
+        pickle.dump(D, g, protocol=4)
+
+    return
+
+
+basename = "xstatpeakc"
+test_num = 0
+np.random.seed(0)
+
+# test with surf['tri']
+rand_surf = {}
+rand_surf["tri"] = np.random.randint(0, int(100 / 2), size=(100, 3))
+
+# iterate the variables
+var_types = {
+    "t": [np.float64],
+    "df": [int, np.uint16],
+    "k": [int, np.uint8],
+    "resl": [np.float64],
+    "tri": [np.int64],
+    "mask": [np.int64],
+}
+
+type_parameter_grid = ParameterGrid(var_types)
+
+for params in type_parameter_grid:
+    slm, thresh, reselspvert, edg = generate_random_peak_clus(rand_surf)
+    for key in list(params.keys()):
+        attr = getattr(slm, key)
+        setattr(slm, key, params[key](attr))
+    test_num += 1
+    params2files(slm, thresh, reselspvert, edg, basename, test_num)
+
+
+for params in type_parameter_grid:
+    slm, thresh, reselspvert, edg = generate_random_peak_clus(
+        rand_surf, reselspvert=True
+    )
+
+    for key in list(params.keys()):
+        attr = getattr(slm, key)
+        setattr(slm, key, params[key](attr))
+    test_num += 1
+    params2files(slm, thresh, reselspvert, edg, basename, test_num)
+
+
+for params in type_parameter_grid:
+    slm, thresh, reselspvert, edg = generate_random_peak_clus(
+        rand_surf, reselspvert=True, edg=True
+    )
+    for key in list(params.keys()):
+        attr = getattr(slm, key)
+        setattr(slm, key, params[key](attr))
+    test_num += 1
+    params2files(slm, thresh, reselspvert, edg, basename, test_num)
+
+
+# test with surf BSPolydata
+pial_fs5 = datasets.fetch_surf_fsaverage()["pial_left"]
+pial_surf = read_surface_gz(pial_fs5)
+n_vertices = get_points(pial_surf).shape[0]
+
+# Variable type tests
+for params in type_parameter_grid:
+    slm, thresh, reselspvert, edg = generate_random_peak_clus(pial_surf)
+    for key in list(params.keys()):
+        attr = getattr(slm, key)
+        setattr(slm, key, params[key](attr))
+    test_num += 1
+    params2files(slm, thresh, reselspvert, edg, basename, test_num)
+
+for params in type_parameter_grid:
+    slm, thresh, reselspvert, edg = generate_random_peak_clus(
+        pial_surf, reselspvert=True
+    )
+    for key in list(params.keys()):
+        attr = getattr(slm, key)
+        setattr(slm, key, params[key](attr))
+    test_num += 1
+    params2files(slm, thresh, reselspvert, edg, basename, test_num)
+
+# Optional variable test.
+var_optional = {
+    "dfs": [None, np.int],
+    "k": [1, 3],
+}
+
+# Additional variable tests.
+additional_parameter_grid = ParameterGrid(var_optional)
+for params in additional_parameter_grid:
+    slm, thresh, reselspvert, edg = generate_random_peak_clus(
+        pial_surf, reselspvert=True
+    )
+    for key in list(params.keys()):
+        setattr(slm, key, params[key])
```

Comment:
This is setting the "dfs" to the class np.int rather than an instance of the class. I'm actually surprised that that _doesn't_ result in an error